### ISSUE TITLE
Batch normalization shared between real and fake examples?

### ISSUE THREAD

**Issue by jiahaowork** [carpedm20/DCGAN-tensorflow#266]
Hi,
It seems that in model.py when discriminator() is called twice, variables of the batch norm layers are also shared between the real and fake example batches. Is this design intentional? But in https://github.com/soumith/ganhacks it suggests that batch norm be separate for the two kinds of examples. Which design is more reasonable? 
Thank you!!
